base/essentials.jl

```diff
@@ -173,6 +173,7 @@ done(v::SimpleVector,i) = (i > v.length)
 isempty(v::SimpleVector) = (v.length == 0)
 indices(v::SimpleVector) = (OneTo(length(v)),)
 linearindices(v::SimpleVector) = indices(v, 1)
+indices(v::SimpleVector, d) = d <= 1 ? indices(v)[d] : OneTo(1)
```

Comment:
does the fallback need to be only for AbstractArray or does it make sense for Any?